File: unity/Assets/Scripts/Multiplayer/Rooms/HeartbeatResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms
{
    /// <summary>
    /// Response returned when a room heartbeat is successfully sent.
    /// Used to maintain the player's presence in the room.
    /// </summary>
    [System.Serializable]
    public class HeartbeatResponse : ApiResponse<ERoomHeartbeatError>
    {
        /// <summary>
        /// Status message indicating the heartbeat was received.
        /// </summary>
        public string status;
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/HeartbeatResponse.java
package com.michitai.multiplayer.rooms;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response returned when a room heartbeat is successfully sent.
 * Used to maintain the player's presence in the room.
 */
public class HeartbeatResponse extends ApiResponse {
    @JsonProperty("status")
    private String status;

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}
